crates/livesplit-auto-splitting/src/process.rs

```diff
@@ -40,166 +44,50 @@ pub type Result<T> = result::Result<T, Error>;
-            CloseHandle(snapshot);
-
-            if let Some((pid, _)) = best_process {
-                Process::with_pid(pid)
-            } else {
-                Err(Error::ProcessDoesntExist)
-            }
+        if let Some((pid, _)) = best_process {
+            Self::with_pid(pid)
+        } else {
+            Err(Error::ProcessDoesntExist)
         }
     }
 
-    pub fn with_pid(pid: DWORD) -> Result<Self> {
-        unsafe {
-            let handle = OpenProcess(PROCESS_VM_READ | PROCESS_QUERY_INFORMATION, false as _, pid);
-
-            if !handle.is_null() {
-                let snapshot = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, pid);
-
-                if snapshot == INVALID_HANDLE_VALUE {
-                    CloseHandle(handle);
-                    return Err(Error::ListModules);
-                }
-
-                let mut modules = HashMap::new();
-                let mut entry: MODULEENTRY32W = mem::uninitialized();
-                entry.dwSize = mem::size_of_val(&entry) as _;
-
-                if Module32FirstW(snapshot, &mut entry) != 0 {
-                    loop {
-                        {
-                            let base_address = entry.modBaseAddr as Address;
-                            let name = &entry.szModule;
-                            let len = name.iter().take_while(|&&c| c != 0).count();
-                            let name = &name[..len];
-                            let name = OsString::from_wide(name).to_string_lossy().into_owned();
-                            modules.insert(name, base_address);
-                        }
-
-                        if Module32NextW(snapshot, &mut entry) == 0 {
-                            break;
-                        }
-                    }
-                }
-
-                let is_64bit;
-                #[cfg(target_pointer_width = "64")]
-                {
-                    use winapi::um::wow64apiset::IsWow64Process;
-
-                    let mut pbool: BOOL = 0;
-                    IsWow64Process(handle, &mut pbool);
-                    is_64bit = pbool == 0;
-                }
-                #[cfg(not(target_pointer_width = "64"))]
-                {
-                    // TODO Actually idk if 32-bit apps can read from 64-bit
-                    // apps. If they can, then this is wrong.
-                    is_64bit = false;
-                }
-
-                CloseHandle(snapshot);
-
-                Ok(Self {
-                    handle,
-                    modules,
-                    is_64bit,
-                })
-            } else {
-                Err(Error::OpenProcess)
-            }
-        }
+    pub fn with_pid(pid: Pid) -> Result<Self> {
+        let modules: HashMap<_,_> = get_process_maps(pid).map_err(|e| Error::ProcessDoesntExist)?
+            .into_iter().filter(|range| range.filename().is_some()).map(|range| {
+                (range.filename().clone().unwrap(), range.start())
+            }).collect();
```

Comment:
should this be written more clearly? i may have a bad habit of cramming lots of iterator chaining into a line or two.

Reply:
I ran rustfmt on the file, which should help.